volatility3/framework/plugins/linux/capabilities.py

```diff
@@ -0,0 +1,228 @@
+# This file is Copyright 2023 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from typing import Iterable, List, Tuple, Dict
+
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.interfaces import plugins
+from volatility3.framework.objects import utility
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.symbols.linux import extensions
+from volatility3.plugins.linux import pslist
+
+vollog = logging.getLogger(__name__)
+
+
+class Capabilities(plugins.PluginInterface):
+    """Lists process capabilities"""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pids",
+                description="Filter on specific process IDs.",
+                element_type=int,
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="inheritable",
+                description="Show only inheritable capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="permitted",
+                description="Show only permitted capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="effective",
+                description="Show only effective capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="bounding",
+                description="Show only bounding capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="ambient",
+                description="Show only ambient capabilities in human-readable strings.",
+                optional=True,
+            ),
+        ]
+
+    def _check_capabilities_support(self):
+        """Checks that the framework supports at least as much capabilities as
+        the kernel being analysed. Otherwise, it shows a warning for the
+        developers.
+        """
+        vmlinux = self.context.modules[self.config["kernel"]]
```

Comment:
There's nothing technically wrong with this, but it's probably better to parameterize the kernel, so that references to `self.config` stay in the main generator/run methods.  This will make it easier to turn into a public classmethod if we want in the future and keeps it in line with most other helper functions.  5:)

Reply:
Actually, this function is not being called from anywhere :) .. I removed the call by mistake, but anyway it doesn't make sense to be called from outside this plugin. The purpose of this function is to warn the developers (us) that they need to update the list of capabilities because it detects that the kernel being analyzed has more capabilities than the ones we support in this plugin. Not sure if you want it here anyway. Thoughts?

Reply:
Errr, yeah, it's fine to keep in here if you'd like.  Happy for you to call it or just leave it as something developers can add in when they need (although that's far less likely to happen), or even to just remove it completely (it'll be in the history now anyway), but if it is going to say, I'd like it to be parameterized (just so people don't quote it to us an as example of core code doing things that way, and therefore they should be able to as well).  5;P